utils/httpx/http.go

```diff
@@ -0,0 +1,87 @@
+package httpx
```

Comment:
this follows the pattern we've used for time, uuids and random numbers - provide a low level utility with easy mocking